client/my-sites/people/people-profile/index.jsx

```diff
@@ -18,8 +21,15 @@ const PeopleProfile = ( { siteId, type, user, invite } ) => {
 	const moment = useLocalizedMoment();
 	const { data: externalContributors } = useExternalContributorsQuery( siteId );
 
-	const getRole = () => {
+	const GetRole = () => {
+		const isPrivate = useSelector( ( state ) => isPrivateSite( state, siteId ) );
```

Comment:
I believe this violates [the rules of hooks](https://reactjs.org/docs/hooks-rules.html) (specifically, "Don’t call Hooks inside loops, conditions, or nested functions."), and renaming `getRole` to `GetRole` [tricks eslint into thinking the function is a react component](https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/src/RulesOfHooks.js#L42-L54), so it's okay to call the selector inside the function.

Moving the call out of the function so it's always called in a predicable order during render should resolve it, and because of the closure mechanism, the value will still be available to the function:

```diff
diff --git a/client/my-sites/people/people-profile/index.jsx b/client/my-sites/people/people-profile/index.jsx
index 26e941c39f..83a395864d 100644
--- a/client/my-sites/people/people-profile/index.jsx
+++ b/client/my-sites/people/people-profile/index.jsx
@@ -21,8 +21,8 @@ const PeopleProfile = ( { siteId, type, user, invite } ) => {
        const moment = useLocalizedMoment();
        const { data: externalContributors } = useExternalContributorsQuery( siteId );
 
+       const isPrivate = useSelector( ( state ) => isPrivateSite( state, siteId ) );
        const GetRole = () => {
-               const isPrivate = useSelector( ( state ) => isPrivateSite( state, siteId ) );
                const wpcomFollowerRole = getWpcomFollowerRole( isPrivate, translate );
 
                if ( invite && invite.role ) {
```

I also recommend moving the name back to the original `getRole`.

Reply:
Today I learned :D Thank you! Will update this.

Reply:
Hooks are definitely a unique feature; there's nothing else really like them in JS, and they're not without some critique in the community.